scripts/perftests/PerformanceTestUtilities.ps1

```diff
@@ -0,0 +1,153 @@
+        { 
+            Write-Host "$($(Get-Date).ToString()): $logStatement"
+        }
+    }
+
+    function GetAbsolutePath([string]$Path)
+    {
+        $Path = [System.IO.Path]::Combine(((pwd).Path), ($Path));
+        $Path = [System.IO.Path]::GetFullPath($Path);
+        return $Path;
+    }
+
+    function OutFileWithCreateFolders([string]$path, [string]$content){
+        $folder = [System.IO.Path]::GetDirectoryName($path)
+        If(!(test-path $folder))
+        {
+            & New-Item -ItemType Directory -Force -Path $folder > $null
+        }
+        Add-Content -Path $path -Value $content
+    }
+
+    function GetAllPackagesInGlobalPackagesFolder([string]$packagesFolder)
+    {
+        if(Test-Path $packagesFolder)
+        {
+            $packages = Get-ChildItem $packagesFolder\*.nupkg -Recurse
+            return $packages
+        }
+        return $null
+    }
+
+    function GetFiles([string]$folder)
+    {
+        if(Test-Path $folder)
+        {
+            $files = Get-ChildItem $folder -recurse
+            return $files
+        }
+        return $null
+    }
+
+    function GetNuGetFoldersPath()
+    {
+        $nugetFolder = [System.IO.Path]::Combine($env:UserProfile, "np")
+        return $nugetFolder
+    }
+
+    function IsClientDotnetExe([string]$nugetClient)
+    {
+        return $nugetClient.endswith("dotnet.exe")
+    }
+
+    function SetupNuGetFolders([string]$_nugetClientFilePath)
+    {
+        $nugetFolders = GetNuGetFoldersPath
+        $Env:NUGET_PACKAGES = [System.IO.Path]::Combine($nugetFolders, "gpf")
+        $Env:NUGET_HTTP_CACHE_PATH = [System.IO.Path]::Combine($nugetFolders, "hcp")
+        $Env:NUGET_PLUGINS_CACHE_PATH = [System.IO.Path]::Combine($nugetFolders, "pcp")
+
+        if($(IsClientDotnetExe $_nugetClientFilePath))
+        {
+            . $_nugetClientFilePath nuget locals -c all  *>>$null
+        } 
+        else 
+        {
+            . $_nugetClientFilePath locals -clear all -Verbosity quiet
+        }
+    }
+
+    function SetupGitRepository([string]$repository, [string]$commitHash, [string]$sourceDirectoryPath)
+    {
+        if(!(Test-Path $sourceDirectoryPath))
+        {
+                git clone $repository $sourceDirectoryPath
+                git -c $sourceDirectoryPath checkout $commitHash
+        }
+        else 
+        {
+                Log "Skipping the cloning of $repository as $sourceDirectoryPath is not empty" -color "Yellow"
+        }
+
+        $gitRepoName = $repository.Substring($($repository.LastIndexOf('/') + 1))
+        $potentialSolutionFile = [System.IO.Path]::Combine($sourceDirectoryPath, "$($gitRepoName.Substring(0, $gitRepoName.Length - 4)).sln")
+
+        if(Test-Path $potentialSolutionFile)
+        {
+            $solutionFile = $potentialSolutionFile
+        } 
+        else 
+        {
+            $solutionFile = (Get-ChildItem $sourceDirectoryPath *.sln)[0] | Select-Object -f 1 | Select-Object -ExpandProperty FullName
```

Comment:
What if this returns `null`? can it be the case that no sln file is found in `$sourceDirectoryPath`?